mypy/types.py

```diff
@@ -1488,3 +1490,49 @@ def true_or_false(t: Type) -> Type:
     new_t.can_be_true = type(new_t).can_be_true
     new_t.can_be_false = type(new_t).can_be_false
     return new_t
+
+
+def replace_none(n: Node) -> Node:
+    """
+    Helper for parse and fastparse.
+    Replace all Nones on the right of protected assignment with Any-like.
```

Comment:
@ddfisher 
Currently mypy complains about assignments like this
```python
x = None  # type: int
```
with ``--strict-optional``. Guido wants this to be changed. I use the term "protected" here with meaning "protected against complains". Since when someone writes the code as above the reason is not to actually assign ``None``, this is just a limitation of the old syntax.

I will add this explanation to the docstring in the next commit.